mantidproject/mantid
# Muon Context

**Description of work.**
This is a simple example of the muon context for the new GUI. Simple group and pair objects have been added to show how to interact with objects (will probably be replaced). The context will contain all of the information for the GUI, but will use objects to minimise the number of members. 

The context file has a list of variable names that correspond to the available members of the context - we will always access the members via the variable names and not the actual strings. 

The example context files demonstrate some functionality. The main features are:
1. The model contains the packing and unpacking of the context - we can then unit test if it is done correctly
2. The context is updated if a display setting changes (e.g. changing a group name will update the pair options)
3. An algorithm is ran (this case we just have a print button)
4. It alters something on a different tab

**To test:**
You will need to add ` Muon/Muon_Analysis_2.py` to the list of interface in the properties file

Open Muon analysis 2. 
Make sure you can also see the mantid logger
Any of the following will cause the current context to be printed: 
     - pressing the print button
     - changing the group names
     - Pressing the buttons on tabs 1 or 3 (these update the text on tab 2).
If the group names update the pair options will also update.



Fixes #23413. 



Does not need to be in the release notes. 
---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/IndividualTicketTesting/)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
